### PR TITLE
Add type check status bar icon. Fixes #57979

### PR DESCRIPTION
**Pull request by nosami**

### CODE REVIEW

**Comment by nosami** [comment]
> otherwise we'll end up with an icon for each languange service

I thought this too... but does it matter?

**Comment by nosami** [comment]
I'll change it anyway.

**Comment by slluis** [comment]
It does matter, we don't want to duplicate logic.

**Comment by nosami** [comment]
OK. Just using the methods that were already available. Not sure why I didn't do that to begin with :)

**Comment by slluis** [comment]
The existing ShowStatusIcon/HideStatusIcon methods can't be reused as is, since they handle some workspace loading events. My proposal was to add new static Show/HideTypeInformationGatheringIcon() methods in TypeSystemService, and have then have MonoDevelopWorkspace and the F# language service use those.

**Comment by nosami** [comment]
Ok, will update

**Comment by nosami** [comment]
Done

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.TypeSystem/MonoDevelopWorkspace.cs [hunk 1]
```diff
@@ -135,29 +132,16 @@ static void OnLoadingFinished (EventArgs e)
...
-				workspacesLoading--;
-				if (workspacesLoading == 0 && statusIcon != null) {
-					statusIcon.Dispose ();
-					statusIcon = null;
-					OnLoadingFinished (EventArgs.Empty);
-					WorkspaceLoaded?.Invoke (this, EventArgs.Empty);
-				}
-			});
+			TypeSystemService.HideTypeInformationGatheringIcon ();
+			OnLoadingFinished (EventArgs.Empty);
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.TypeSystem/MonoDevelopWorkspace.cs]: We now no longer trigger this event on the UI thread.

**Comment by nosami** [comment]
@Therzok Good catch, fixed

**Pull request merged**